.github/workflows/test.yml

```diff
@@ -1,24 +1,30 @@
 name: Test
 
 on:
-  - pull_request
-  - push
+  push:
+    branches: ["**"]
+  pull_request:
+    branches: ["**"]
 
 jobs:
   test:
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        ruby:
-          - "2.7"
-          - "3.0"
+        ruby: ["2.7", "3.0", "3.1", "3.2", "3.3"]
+
     steps:
       - uses: actions/checkout@v4
-      - name: Install Ruby ${{ matrix.ruby }}
+
+      - name: Setup Ruby ${{ matrix.ruby }}
         uses: ruby/setup-ruby@v1
         with:
           ruby-version: ${{ matrix.ruby }}
           bundler-cache: true
+
+      - name: Install dependencies
```

Comment:
`bundler-cache` above will trigger a `bundle install && bundle update`, you don't need this step.